jupyterhub/handlers/base.py

```diff
@@ -467,36 +533,55 @@ def get_next_url(self, user=None):
                 next_url = url_path_join(self.hub.base_url, 'home')
         return next_url
 
+    async def auth_to_user(self, authenticated, user=None):
+        """Persist data from .authenticate() or .refresh_user() to the User database
+
+        Args:
+            authenticated(dict): return data from .authenticate or .refresh_user
+            user(User, optional): the User object to refresh, if refreshing
+        Return:
+            user(User): the constructed User object
+        """
+        if isinstance(authenticated, str):
```

Comment:
`.authenticate()` returning just a string is the old interface right? Is there a plan to retire it? If yes maybe we should create an issue to come up with a plan and a helper for this pattern that also produces a warning?

Reply:
I don't think we should ever drop support for authenticate returning a string, so this should always work without warning. That's the simplest way to implement an authenticator, and is all you need in most cases.